library/ssl_tls.c

```diff
@@ -3643,6 +3643,55 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     {
         ssl->handshake = mbedtls_calloc( 1, sizeof(mbedtls_ssl_handshake_params) );
     }
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    /* If the buffers are too small - reallocate */
+    {
+        int modified = 0;
+        unsigned char *resized_buffer = NULL;
+        if( ssl->in_buf_len < MBEDTLS_SSL_IN_BUFFER_LEN )
+        {
+            resized_buffer = mbedtls_calloc( 1, MBEDTLS_SSL_IN_BUFFER_LEN );
+            if( resized_buffer != NULL )
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 2, ( "Reallocating in_buf to %d", MBEDTLS_SSL_IN_BUFFER_LEN ) );
+                memcpy( resized_buffer, ssl->in_buf, ssl->in_buf_len );
+                mbedtls_platform_zeroize( ssl->in_buf, ssl->in_buf_len );
+                mbedtls_free( ssl->in_buf );
+
+                ssl->in_buf = resized_buffer;
```

Comment:
Does LCOV show we reached here in our unit testing?

Reply:
Yes.